backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -81,4 +81,28 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE, "Tried to fetch tokens of a shop which is not in the ACTIVE state.");
 		return new ArrayList<>();
 	}
+
+	public Token addTokenToShop(UUID shopId) {
+		if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+			Integer lastToken = shopIdToLastAllotedTokenMap.get(shopId).incrementAndGet();
+			UUID uuid = UUID.randomUUID();
+			Token newToken = new Token();
+			newToken.setTokenId(uuid);
+			newToken.setShopId(shopId);
+			newToken.setTokenNumber(lastToken);
+			newToken.setStatus(Token.Status.ACTIVE);
+			tokenMap.put(uuid, newToken);
+			if (lastToken != 1) {
+				List<Token> newTokenList = shopIdToListOfTokensMap.get(shopId);
+				newTokenList.add(newToken);
+				shopIdToListOfTokensMap.put(shopId, newTokenList);
+			} else {
+				List<Token> newTokenList = new ArrayList<>();
+				newTokenList.add(newToken);
+				shopIdToListOfTokensMap.put(shopId, newTokenList);
+			}
+			return newToken;
+		}
+		return new Token();
```

Comment:
Add a comment "TODO: throw exception here". above this.

Reply:
Done